spec/lib/synced/model_spec.rb

```diff
@@ -166,6 +118,41 @@ def self.column_names
         Booking.synchronize(scope: account, query_params: { from: from })
       end
     end
+
+    context "scope" do
+      let(:account) { Account.create(name: "test") }
+      let(:strategy) { double(:strategy, perform: true) }
+
+      context "when associaiton is present" do
+        it "finds scope from the association" do
+          expect(Synced::Synchronizer).to receive(:new).with(anything, hash_including(scope: account)).and_return(strategy)
```

Comment:
`.with(anything, hash_including(scope: account))` - this is some hardcore level witchcraft that this even works, but at least it is expressive :P. Good to know that it is possible.